File: client/src/event.ts
// Copyright 2020 Tim Shannon. All rights reserved.
// Use of this source code is governed by the MIT license
// that can be found in the LICENSE file.

// fields need to match up with event.go
export class GoEvent {
    public type: string;
    public altKey: boolean;
    public button: number;
    public buttons: number;
    public clientX: number;
    public clientY: number;
    public ctrlKey: boolean;
    public metaKey: boolean;
    public movementX: number;
    public movementY: number;
    public screenX: number;
    public screenY: number;
    public shiftKey: boolean;
    constructor(e: any) {
        this.type = e.type;
        this.altKey = e.altKey;
        this.button = e.button;
        this.buttons = e.buttons;
        this.clientX = e.clientX;
        this.clientY = e.clientY;
        this.ctrlKey = e.ctrlKey;
        this.metaKey = e.metaKey;
        this.movementX = e.movementX;
        this.movementY = e.movementY;
        this.screenX = e.screenX;
        this.screenY = e.screenY;
        this.shiftKey = e.shiftKey;
    }
}

File: event.go
// Copyright 2020 Tim Shannon. All rights reserved.
// Use of this source code is governed by the MIT license
// that can be found in the LICENSE file.

package reflex

import "reflect"

// Event is a DOM event optionally available as a parameter to EventFuncs
type Event struct {
	Type string
	MouseEvent
}

var eventType = reflect.TypeOf(Event{})

// MouseEvent is the base structure from a DOM mouse event
type MouseEvent struct {
	AltKey    bool `json:"altKey"`
	Button    uint `json:"button"`
	Buttons   uint `json:"buttons"`
	ClientX   uint `json:"clientX"`
	ClientY   uint `json:"clientY"`
	CtrlKey   bool `json:"ctrlKey"`
	MetaKey   bool `json:"metaKey"`
	MovementX uint `json:"movementX"`
	MovementY uint `json:"movementY"`
	ScreenX   uint `json:"screenX"`
	ScreenY   uint `json:"screenY"`
	ShiftKey  bool `json:"shiftKey"`
}
